### ISSUE TITLE
Document what exactly --backend-aggressive-opts does

### ISSUE THREAD

**Issue by KOLANICH** [avast/retdec#1016]
When I have it enabled sometimes I see a lot of garbage (well, it may be not garbage, but when the flag is not set, these are not emitted)  in the beginnings of functions (instead of reduction of amount of garbage that I expect).

Also it seems to be eliminating some expressions, leaving behind an emty uninitialized variable (which was initialized by the eliminated expression), that is used later.

In other words, currently this option seems to be completely useless and harmful.

**Comment by PeterMatula** [comment]
It took some investigation, but it looks like it enables any optimization that is prefixed with `Aggressive` (in its name in source codes). Currently, it means these two:
* [AggressiveDerefOptimizer](https://github.com/avast/retdec/blob/ea28eab22e6093a28e5826877e5ea87df33dfe78/include/retdec/llvmir2hll/optimizer/optimizers/aggressive_deref_optimizer.h)
* [AggressiveGlobalToLocalOptimizer](https://github.com/avast/retdec/blob/ea28eab22e6093a28e5826877e5ea87df33dfe78/include/retdec/llvmir2hll/optimizer/optimizers/aggressive_global_to_local_optimizer.h)

Looking at those comments, they look like antiquated ideas that we implemented a long time ago and never properly used. Also looking at them, they don't look like a good idea - both are indeed harmful and we should not do them if we don't want to damage the original semantics.

I will remove them and all the related code momentarily.

**Comment by KOLANICH** [comment]
Thanks.
